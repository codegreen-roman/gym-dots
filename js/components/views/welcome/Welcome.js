import React, { Component } from 'react'
import { func, object } from 'prop-types'
import { equals } from 'ramda'
import {
    welcomeSection,
    welcomeHeader,
    welcomeBody,
    buttonsWrapper,
    buttonAndIconWrap,
    buttonText,
    headerDate,
    headerText
} from './Welcome.glamor.js'
import { Icon } from '@components/common/icon/Icon'
import { variables } from '@utils/variables'

export class Welcome extends Component {

    componentWillReceiveProps({status, history}) {
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            history.push('/activity/pre')
        }
    }

    render() {
        const { status, loginGuest, loginWith, dateStr} = this.props
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            return null
        }

        return (
            <section {...welcomeSection}>
                <div {...welcomeHeader}>
                    <div {...headerDate} data-test='date'>{dateStr}</div>
                    <div {...headerText}>Gym dots</div>
                </div>
                <div {...welcomeBody}>
                    <div>Welcome!</div>
                    <div>Login into gym dots systems</div>
                    <div {...buttonsWrapper}>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='facebook' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='facebook' onClick={() => loginWith('facebook')}>login with facebook</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='twitter' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='twitter' onClick={() => loginWith('twitter')}>login with twitter</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='guest' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='guest' onClick={loginGuest}>login as guest</button>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}

Welcome.propTypes = {
    loginWith: func.isRequired,
    loginGuest: func.isRequired,
    history: object
}
