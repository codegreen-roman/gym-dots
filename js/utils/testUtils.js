export const findByDataAttr = (wrapper, value) => {
    return wrapper.find(`[data-test='${value}']`)
}
