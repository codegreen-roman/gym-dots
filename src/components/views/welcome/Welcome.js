import React, { PureComponent } from 'react'
import { func, object, string } from 'prop-types'
import { equals } from 'ramda'
import {
    welcomeSection,
    welcomeHeader,
    welcomeBody,
    buttonsWrapper,
    buttonAndIconWrap,
    buttonText,
    headerDate,
    headerText
} from './Welcome.glamor.js'
import { Icon } from '@components/common/icon/Icon'
import { variables } from '@utils/variables'

export class Welcome extends PureComponent {

    static propTypes = {
        loginWith: func.isRequired,
        status: string.isRequired,
        dateStr: string.isRequired,
        loginGuest: func.isRequired,
        history: object
    }

    componentWillReceiveProps({status, history}) {
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            history.push('/in/activity/pre')
        }
    }

    render() {
        const { status, loginGuest, loginWith, dateStr } = this.props
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            return null
        }

        return (
            <section {...welcomeSection}>
                <div {...welcomeHeader}>
                    <div {...headerDate} data-test='date'>{dateStr}</div>
                    <div {...headerText}>Gym dots</div>
                </div>
                <div {...welcomeBody}>
                    <div>Welcome!</div>
                    <div>Login into gym dots systems</div>
                    <div {...buttonsWrapper}>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='facebook'
                                color={variables.$color2}
                                width={26}
                                height={26}
                            />
                            <button {...buttonText} data-test='facebook' onClick={() => loginWith('facebook')}>login with facebook</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='twitter'
                                color={variables.$color2}
                                viewBox='0 0 20 17'
                                width={23}
                                height={23}
                            />
                            <button {...buttonText} data-test='twitter' onClick={() => loginWith('twitter')}>login with twitter</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='guest'
                                color={variables.$color2}
                                width={26}
                                height={26}
                            />
                            <button {...buttonText} data-test='guest' onClick={loginGuest}>login as guest</button>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}
