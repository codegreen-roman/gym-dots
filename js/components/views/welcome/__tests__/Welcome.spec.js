import React from 'react'
import { Welcome } from '../Welcome'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { findByDataAttr } from '@utils/testUtils'

const defaultProps = {
    dateStr: 'Saturday, Jul 15th',
    exerciseName: '',
    status: '',
    loginWith: jest.fn(),
    loginGuest: jest.fn()
}

const setup = (props = defaultProps) => {
    return shallow(<Welcome {...props} />)
}

describe('Welcome component', () => {

    describe('Not logged in', () => {

        const props = {
            dateStr: 'Saturday, Jul 15th',
            status: 'loggedOut',
            loginWith: jest.fn(),
            loginGuest: jest.fn()
        }

        let wrapper
        let twitterBtn
        let facebookBtn
        let guestBtn
        let dateElement
        const currentDate = 'Saturday, Jul 15th'

        beforeEach(() => {
            wrapper = setup(props)
            twitterBtn = findByDataAttr(wrapper, 'twitter')
            facebookBtn = findByDataAttr(wrapper, 'facebook')
            guestBtn = findByDataAttr(wrapper, 'guest')
            dateElement = findByDataAttr(wrapper, 'date')
        })

        describe('Should contain', () => {

            test('dateElement with a date value Saturday, Jul 15th', () => {
                expect(dateElement.text()).toBe(currentDate)
            })

            test('twitterBtn with text', () => {
                expect(twitterBtn.text()).toBe('login with twitter')
            })

            test('facebookBtn with text', () => {
                expect(facebookBtn.text()).toBe('login with facebook')
            })

            test('guestBtn with text', () => {
                expect(guestBtn.text()).toBe('login as guest')
            })
        })

        describe('When performing login', () => {

            test('should call the loginWith with twitter provider', () => {
                twitterBtn.simulate('click')
                expect(props.loginWith).toBeCalledWith('twitter')
            })

            test('should call the loginWith with facebook provider', () => {
                facebookBtn.simulate('click')
                expect(props.loginWith).toBeCalledWith('facebook')
            })

            test('should call the loginGuest', () => {
                guestBtn.simulate('click')
                expect(props.loginGuest).toHaveBeenCalled()
            })
        })

    })

    describe('Logged in', () => {

        const props = {
            dateStr: 'Saturday, Jul 15th',
            status: 'loggedIn',
            loginWith: jest.fn(),
            loginGuest: jest.fn()
        }

        let wrapper
        let twitterBtn
        let facebookBtn
        let guestBtn

        beforeEach(() => {
            wrapper = setup(props)
            twitterBtn = findByDataAttr(wrapper, 'twitter')
            facebookBtn = findByDataAttr(wrapper, 'facebook')
            guestBtn = findByDataAttr(wrapper, 'guest')
        })

        test('should not have login buttons', () => {
            expect(twitterBtn.length).toBe(0)
            expect(facebookBtn.length).toBe(0)
            expect(guestBtn.length).toBe(0)
        })

    })

    describe('Shallow snapshot', () => {
        test('matches the previous', () => {
            const component = setup()
            const tree = shallowToJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})
