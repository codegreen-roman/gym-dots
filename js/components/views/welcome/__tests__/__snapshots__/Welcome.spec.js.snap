// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component Shallow snapshot matches the previous 1`] = `
<section
  data-css-kj33m7=""
>
  <div
    data-css-ut9bv=""
  >
    <div
      data-css-1ixk24o=""
      data-test="date"
    >
      Saturday, Jul 15th
    </div>
    <div
      data-css-1qe4ilz=""
    >
      Gym dots
    </div>
  </div>
  <div
    data-css-5vm3an=""
  >
    <div>
      Welcome!
    </div>
    <div>
      Login into gym dots systems
    </div>
    <div
      data-css-1frxcgx=""
    >
      <div
        data-css-16vjom5=""
      >
        <Icon
          color="#5E6B7D"
          icon="facebook"
          size={30}
        />
        <button
          data-css-3l7lts=""
          data-test="facebook"
          onClick={[Function]}
        >
          login with facebook
        </button>
      </div>
      <div
        data-css-16vjom5=""
      >
        <Icon
          color="#5E6B7D"
          icon="twitter"
          size={30}
        />
        <button
          data-css-3l7lts=""
          data-test="twitter"
          onClick={[Function]}
        >
          login with twitter
        </button>
      </div>
      <div
        data-css-16vjom5=""
      >
        <Icon
          color="#5E6B7D"
          icon="guest"
          size={30}
        />
        <button
          data-css-3l7lts=""
          data-test="guest"
          onClick={[Function]}
        >
          login as guest
        </button>
      </div>
    </div>
  </div>
</section>
`;
